File: backend/src/schemas/support.py
from enum import Enum

from pydantic import BaseModel, EmailStr, Field


class SupportRequestType(str, Enum):
    ACCOUNT_CREATION = "account_creation"
    ACCESS_ISSUE = "access_issue"
    GENERAL = "general"


class SupportContactRequest(BaseModel):
    name: str = Field(min_length=1, max_length=120)
    email: EmailStr
    request_type: SupportRequestType = SupportRequestType.ACCOUNT_CREATION
    message: str = Field(min_length=10, max_length=4000)
    restaurant_name: str | None = Field(default=None, max_length=200)
    source: str = Field(default="web", max_length=32)


class SupportContactResponse(BaseModel):
    message: str


class SupportInfoResponse(BaseModel):
    contact_email: str
    onboarding_steps: list[str]

File: frontend/web/src/api/Support.ts
import { apiClient } from './API';

export type SupportRequestType = 'account_creation' | 'access_issue' | 'general';

export interface SupportContactPayload {
  name: string;
  email: string;
  request_type: SupportRequestType;
  message: string;
  restaurant_name?: string;
  source?: string;
}

export interface SupportInfo {
  contact_email: string;
  onboarding_steps: string[];
}

export const getSupportInfo = () =>
  apiClient.get<SupportInfo>('/api/support/info').then((res) => res.data);

export const submitSupportRequest = (payload: SupportContactPayload) =>
  apiClient.post<{ message: string }>('/api/support/contact', payload).then((res) => res.data);
